p2p/ping.go

```diff
@@ -79,50 +79,53 @@ func pingPeer(ctx context.Context, svc *ping.PingService, p peer.ID, callback fu
 	}
 }
 
-// pingPeerOnce starts a long lived ping connection with the peer and returns on first error.
+// pingPeerOnce starts a long-lived ping connection with the peer and returns on first error.
 func pingPeerOnce(ctx context.Context, svc *ping.PingService, p peer.ID,
```

Comment:
all these closures and callbacks and stuff is very confusing, lets try and simplify into a simple loop. Take a step back, zoom out, think what the problem is, then solve it from scratch, we are building on top of previous stuff, we are digging the hole deeper and deeper